### PR TITLE
Add unit tests for Two_Factor_Provider::get_code().

### PR DESCRIPTION
**Pull request by dd32**
This was split out of #518 as it's technically unrelated, and is failing for some reason in GitHub actions, but passing locally.

I can't really determine by looking at the code why it's failing though.

### CODE REVIEW

**Comment by dd32** [comment]
This fails due to https://core.trac.wordpress.org/ticket/55194

Opened #523 to run unit tests against trunk as well.

**Comment by dd32** [comment]
In making this testable, since `Two_Factor_Provider` is an abstract class, I chose to simply make it a static method due to it's lack of `$this->` usage. `$this->get_code()` is still acceptable, but now `Two_Factor_Provider::get_code()` also is.

File: tests/providers/class-two-factor-provider.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	 */
+	function test_get_code() {
+		$code = Two_Factor_Provider::get_code( 3, '1' );
+		$this->assertEquals( '111', $code );
+
+		$code = Two_Factor_Provider::get_code( 8, '1' );
+		$this->assertEquals( '11111111', $code );
+
+		$code = Two_Factor_Provider::get_code( 8, 'A' );
+		$this->assertEquals( 'AAAAAAAA', $code );
```
Comment (iandunn) [inline tests/providers/class-two-factor-provider.php:26]:
```suggestion
		$this->assertEquals( 'AAAAAAAA', $code );
		
		$code = Two_Factor_Provider::get_code( 30, array( 'A', 'B', 'C' ) );
		$this->assertSame( 1, preg_match( '/^[ABC]*$/', $code ) );
```

This tests the `$chars = implode( '', $chars );` statement.

**Review by iandunn — APPROVED** [review]
Looks like this is working after #524 👍🏻

File: tests/providers/class-two-factor-provider.php (hunk 1 above)
Comment (dd32) [reply tests/providers/class-two-factor-provider.php:26]: Good call, I extended that to also validate that the generated code is 30char as requested, and an additional test that validates non-1-char-length strings.

Comment (iandunn) [reply tests/providers/class-two-factor-provider.php:26]: 👍🏻

**Pull request merged**